circuit_maintenance_parser/parsers/cogent.py

```diff
@@ -0,0 +1,110 @@
+"""Cogent parser."""
+import logging
+import re
+from typing import Dict
+from datetime import datetime
+from pytz import timezone, UTC
+from bs4.element import ResultSet  # type: ignore
+
+from circuit_maintenance_parser.errors import ParsingError
+from circuit_maintenance_parser.parser import Html, Impact, CircuitImpact, Status
+from circuit_maintenance_parser.utils import city_timezone
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=too-many-branches
+
+
+class ParserCogent(Html):
+    """Notifications Parser for Cogent notifications."""
+
+    provider_type: str = "cogent"
+
+    # Default values for cogent notifications
+    _default_provider = "cogent"
+    _default_organizer = "support@cogentco.com"
+
+    def parse_html(self, soup, data_base):
+        """Execute parsing."""
+        data = data_base.copy()
+        try:
+            self.parse_div(soup.find_all("div", class_="a3s aiL"), data)
+            self.parse_title(soup.find_all("title"), data)
+            return [data]
+
+        except Exception as exc:
+            raise ParsingError from exc
+
+    def parse_div(self, divs: ResultSet, data: Dict):  # pylint: disable=too-many-locals
+        """Parse <div> tag."""
+        start_str = ""
+        end_str = ""
+
+        for div in divs:
+            for line in div.text.splitlines():
+                if line.endswith("Network Maintenance"):
+                    data["summary"] = line
+                elif line.startswith("Dear"):
+                    match = re.search(r"Dear (.*),", line)
+                    if match:
+                        data["account"] = match.group(1)
+                elif line.startswith("Start time:"):
+                    match = re.search(r"Start time: (.*) \([A-Za-z\s]+\) (\d+/\d+/\d+)", line)
+                    if match:
+                        start_str = " ".join(match.groups())
+                elif line.startswith("End time:"):
+                    match = re.search(r"End time: (.*) \([A-Za-z\s]+\) (\d+/\d+/\d+)", line)
+                    if match:
+                        end_str = " ".join(match.groups())
+                elif line.startswith("Cogent customers receiving service"):
+                    match = re.search(r"[^Cogent].*?((\b[A-Z][a-z\s-]+)+, ([A-Za-z-]+[\s-]))", line)
+                    if match:
+                        parsed_timezone = city_timezone(match.group(1).strip())
+                        local_timezone = timezone(parsed_timezone)
+                        # set start time using the local city timezone
+                        start = datetime.strptime(start_str, "%I:%M %p %d/%m/%Y")
```

Comment:
maybe I'm missing something, but what if the line processing for `start_str` happens later and here we still have `start_str=""`? should not  be this done in different loops?

Reply:
maybe you could wrap up everything in a more complete function such as:
```py
def convert_time_to_utc(timezone, time):
    parsed_timezone = city_timezone(match.group(1).strip())
    local_timezone = timezone(parsed_timezone)
    local_time = local_timezone.localize(time)
    return local_time.astimezone(UTC)
# Maybe also add the logging here 
```

Reply:
> maybe I'm missing something, but what if the line processing for `start_str` happens later and here we still have `start_str=""`? should not be this done in different loops?

but I see that the email format is as it is, so it happens later